flink-libraries/flink-table/pom.xml

```diff
@@ -252,6 +261,29 @@ under the License.
 									<pattern>org.eigenbase</pattern>
 									<shadedPattern>org.apache.flink.shaded.calcite.org.eigenbase</shadedPattern>
 								</relocation>
+								<relocation>
+									<pattern>org.pentaho</pattern>
+									<shadedPattern>org.apache.flink.shaded.calcite.org.pentaho</shadedPattern>
+								</relocation>
+
+								<!-- flink-table dependencies -->
+								<relocation>
+									<pattern>org.apache.commons</pattern>
+									<shadedPattern>org.apache.flink.shaded.org.apache.commons</shadedPattern>
+								</relocation>
+								<relocation>
+									<pattern>org.reflections</pattern>
+									<shadedPattern>org.apache.flink.shaded.org.reflections</shadedPattern>
+								</relocation>
+								<relocation>
+									<pattern>org.joda.time</pattern>
+									<shadedPattern>org.apache.flink.shaded.org.joda.time</shadedPattern>
+								</relocation>
+								<!-- not relocated for now -->
```

Comment:
why not?

Reply:
The problem is that Calcite is using the Janino properties file for determining the location of the compiler. We could relocate Janino as well but then we also have to change the contents of the properties file. Have you done this before?

Reply:
no, but that's exactly what you should write as a comment into the pom ;)